src/plugins/ds_mongo.c

```diff
@@ -515,6 +537,117 @@ srpds_get_predicate(const struct lyd_node *node, const char **predicate, char **
     return err_info;
 }
 
+/**
+ * @brief Iterates through @p path in the direction @p direction ,
+ *      finds symbol @p symbol and returns a pointer to the symbol.
+ *
+ * @param[in] path Path ending with NULL.
+ * @param[in] symbol Symbol to find.
+ * @param[in] direction Direction in which to go through @p path .
```

Comment:
It says nothing about `path` being expected to point to the end when `direction` is `-1` and I do not think there should be this expectation, it can be moved at the end in the function.